libparsec/crates/protocol/tests/authenticated_cmds/v3/invite_new.rs

```diff
@@ -1,27 +1,147 @@
+            // Content:
+            //   status: "ok"
+            //   token: ext(2, hex!("d864b93ded264aae9ae583fd3d40c45a"))
+            //
+            // Note that raw data does not contain "email_sent".
+            // This was valid behavior in api v2 but is no longer valid from v3 onwards.
+            // The corresponding expected values used here are therefore not important
+            // since loading raw data should fail.
+            //
+            &hex!("82a6737461747573a26f6ba5746f6b656ed802d864b93ded264aae9ae583fd3d40c45a")[..],
+            Err("missing field `email_sent`".into()),
+        ),
+        (
+            // Generated from Python implementation (Parsec v2.6.0+dev)
+            // Content:
+            //   email_sent: "SUCCESS"
+            //   status: "ok"
+            //   token: ext(2, hex!("d864b93ded264aae9ae583fd3d40c45a"))
+            &hex!(
+                "83aa656d61696c5f73656e74a753554343455353a6737461747573a26f6ba5746f6b656ed8"
+                "02d864b93ded264aae9ae583fd3d40c45a"
+            )[..],
+            Ok(authenticated_cmds::invite_new::Rep::Ok {
+                token: InvitationToken::from_hex("d864b93ded264aae9ae583fd3d40c45a").unwrap(),
+                email_sent: authenticated_cmds::invite_new::InvitationEmailSentStatus::Success,
+            }),
+        ),
+    ];
+
+    for (raw, expected) in raw_expected {
+        rep_helper(raw, expected);
+    }
 }
 
 pub fn rep_not_allowed() {
-    // TODO #4545: Implement test
+    // Generated from Python implementation (Parsec v2.6.0+dev)
+    // Content:
+    //   status: "not_allowed"
+    let raw = hex!("81a6737461747573ab6e6f745f616c6c6f776564");
+    let expected = Ok(authenticated_cmds::invite_new::Rep::NotAllowed);
+    rep_helper(&raw, expected);
 }
 
 pub fn rep_already_member() {
-    // TODO #4545: Implement test
+    // Generated from Python implementation (Parsec v2.6.0+dev)
+    // Content:
+    //   status: "already_member"
+    let raw = hex!("81a6737461747573ae616c72656164795f6d656d626572");
+    let expected = Ok(authenticated_cmds::invite_new::Rep::AlreadyMember);
+    rep_helper(&raw, expected);
 }
 
 pub fn rep_not_available() {
-    // TODO #4545: Implement test
+    // Generated from Python implementation (Parsec v2.6.0+dev)
+    // Content:
+    //   status: "not_available"
+    let raw = hex!("81a6737461747573ad6e6f745f617661696c61626c65");
+    let expected = Ok(authenticated_cmds::invite_new::Rep::NotAvailable);
+    rep_helper(&raw, expected)
+}
+
+fn rep_helper(raw: &[u8], expected: Result<authenticated_cmds::invite_new::Rep, String>) {
+    let res = authenticated_cmds::invite_new::Rep::load(raw).map_err(|e| e.to_string());
+
+    p_assert_eq!(res, expected);
+
+    if let Ok(data) = (res) {
+        // Also test serialization round trip
+        let raw2 = data.dump().unwrap();
+
+        let data2 = authenticated_cmds::invite_new::Rep::load(&raw2).map_err(|e| e.to_string());
```

Comment:
don't map to string

Reply:
`::rpm_serde::decode::Error` don't implement `PartialEq` & `Eq` do you have a suggestion to circumvent that ?

Reply:
Don't group error case with good case

Reply:
done